src/app/modules/item/components/thread/thread.component.ts

```diff
@@ -24,16 +24,13 @@ export class ThreadComponent implements AfterViewInit, OnDestroy {
   constructor(
     private threadService: ThreadService,
     private fb: FormBuilder,
-    private userSessionService: UserSessionService,
   ) {}
 
   ngAfterViewInit(): void {
-    this.subscription = combineLatest([
-      this.state$.pipe(readyData()),
-      this.userSessionService.userProfile$,
-    ]).subscribe(([ events, currentUser ]) => {
-      const lastEvent = events[events.length - 1];
-      if (lastEvent && lastEvent.createdBy === currentUser.groupId) {
+    this.subscription = this.state$.pipe(readyData()).subscribe(() => {
+      if (this.messagesScroll && this.sendMessageForm && (this.messagesScroll.nativeElement.scrollHeight
+        <= (this.messagesScroll.nativeElement.scrollTop + this.messagesScroll.nativeElement.offsetHeight
+        + this.sendMessageForm.nativeElement.offsetHeight + parseInt(getComputedStyle(this.messagesScroll.nativeElement).paddingBottom)))) {
         setTimeout(() => {
           this.scrollDown();
         });
```

Comment:
This `setTimeout(() => {}` is like putting a subscribe in a subscribe.. or at least mixing Rx with another async technique.
I think this is somehow equivalent to:
```
this.subscription = this.state$.pipe(
  filter(s => s.isReady),
  filter(() => this.messagesScroll && this.se ... ment).paddingBottom)),
  delay(0),
).subscribe(() => {
  this.scrollDown();
});

Reply:
Fixed